.github/workflows/docker-image.yml

```diff
@@ -1,26 +1,31 @@
 name: Docker Image CI
 on:
   push:
-    branches: ['master']
+    branches:
+      - master
+  pull_request:
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
     - name: Login in quay
+      if: github.ref == 'refs/heads/master' && github.event_name == 'push'
       run: docker login quay.io -u ${QUAY_USER} -p ${QUAY_TOKEN}
```

Comment:
I'm not 100% sure on this but if there's a PR from a fork I do not think this will work with the user and token coming from secrets. We hit a lot of restrictions on what we could and couldn't do in github actions because our process includes forking

Reply:
Might need to create a PR from a fork as well to verify this works both ways. Again not sure this would work and pick up the updates made from this PR. Might have to merge and then retest on fork

Reply:
Yes, you're right - in this case and related steps, I've added an if condition that stops this step from running from a fork - it will only run with push events into the main branch, such as when a PR is merged.

This only enables us to get the other basic checks we need for PRs so that we don't have to run these steps manually.